lib/Sema/CSSimplify.cpp

```diff
@@ -950,7 +950,6 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
   // Apply labels to arguments.
   SmallVector<AnyFunctionType::Param, 8> argsWithLabels;
   argsWithLabels.append(args.begin(), args.end());
-  AnyFunctionType::relabelParams(argsWithLabels, argLabels);
```

Comment:
I'd suggest we leave this be for now since this is not related to the problem.

Reply:
Will do. What about the assertion?

Reply:
I think that assertions makes more sense in SILGen maybe to make sure that number of indexHashes retrieved matches the number of parameters...

Reply:
I mean, what should we do about [the assertion](https://github.com/apple/swift/blob/5f3b1da22a3d0b2f2d90540cbbd8336db7ca931c/lib/AST/ASTContext.cpp#L2982) that is triggered in `relabelParams`?

Reply:
An alternative solution would be to update the subscript component in CSApply with a new list of argument labels to include the default arguments. We already have to [iterate over the parameters to dig out the conformances here](https://github.com/apple/swift/blob/4980bc71ee96649c3ac835c123fb10a89b7f0689/lib/Sema/CSApply.cpp#L4646-L4659), so we could also use that loop to build up a new vector of argument labels.

Eventually, we should probably make `coerceCallArguments` return the individually coerced arguments along with their labels instead of bundling them into a paren or tuple expr, at which point we could use the labels returned from that.

Reply:
I like what @hamishknight is proposing, since `coerceCallArguments` could produce defaults we can't actually rely on the label array from subscript component.

Reply:
Sounds good, I'll work on that.

Reply:
Turns out I made a mistake. This isn't actually an issue. I misdiagnosed something earlier. I won't change anything here but, maybe I will make another patch to update `coerceCallArguments` as you suggested, @hamishknight.